Kernel/Net/Routing.cpp

```diff
@@ -1,10 +1,124 @@
-#include <Kernel/Net/LoopbackAdapter.h>
 #include <Kernel/Net/Routing.h>
+#include <Kernel/Thread.h>
 
-WeakPtr<NetworkAdapter> adapter_for_route_to(const IPv4Address& ipv4_address)
+//#define ROUTING_DEBUG
+
+Lockable<HashMap<IPv4Address, MACAddress>>& arp_table()
 {
-    // FIXME: Have an actual routing table.
-    if (ipv4_address == IPv4Address(127, 0, 0, 1))
-        return LoopbackAdapter::the().make_weak_ptr();
-    return NetworkAdapter::from_ipv4_address(IPv4Address(192, 168, 5, 2));
+    static Lockable<HashMap<IPv4Address, MACAddress>>* the;
+    if (!the)
+        the = new Lockable<HashMap<IPv4Address, MACAddress>>;
+    return *the;
+}
+
+RoutingDecision route_to(const IPv4Address& target, const IPv4Address& source)
+{
+    auto target_addr = target.to_u32();
+    auto source_addr = source.to_u32();
+
+    WeakPtr<NetworkAdapter> local_adapter = nullptr;
```

Comment:
I suspect that we're heading in the wrong direction trying to use `WeakPtr` for network adapters.
Given that hot-plugging is a thing, it seems like a real concern that these pointers would become null in the middle of some logic.

Maybe we could just keep the adapter objects alive but turn everything into no-ops, and then tear them down at some safe checkpoints?
Anyways, that's not something you have to fix in this patch.